### PR TITLE
Added db_last_error.php to install check, corrected auto load of jQuery

### LINKED ISSUE #3114: Install database error for missing db_last_error.php
Warning: file_put_contents(/smf/db_last_error.php): failed to open stream: Permission denied in /smf/Settings.php on line 195
1. We should include this file (empty) by default in our installer.
2. We should make sure on install/upgrade that this file exists and is writable.

### PR DESCRIPTION
**Pull request by tinoest**
Fixes #3114 

Signed-off-by: tinoest tinoest@gmail.com

### CODE REVIEW

**Comment by Oldiesmann** [comment]
Don't we need to make other changes to support jQuery 3.1.1?

**Comment by tinoest** [comment]
Not that I can see, the rest were done which I could find in my previous pull request #3807 which has already been merged.

File: Sources/ManageMaintenance.php [hunk 1]
```diff
@@ -1725,7 +1725,7 @@ function MaintainRecountPosts()
 	// Lets get a group of members and determine their post count (from the boards that have post count enabled of course).
 	$request = $smcFunc['db_query']('', '
 		SELECT /*!40001 SQL_NO_CACHE */ m.id_member, COUNT(m.id_member) AS posts
-		FROM ({db_prefix}messages AS m, {db_prefix}boards AS b)
+		FROM {db_prefix}messages AS m, {db_prefix}boards AS b
```
Comment (jdarwood007) [inline Sources/ManageMaintenance.php]: Wait when was this done?

We don't use multiple from statements.  We use joins to bring in additional tables to make it clear what we are doing.  http://stackoverflow.com/questions/894490/sql-left-join-vs-multiple-tables-on-from-line

**Comment by tinoest** [comment]
No idea, I just corrected it so it ran on PG aswell as MySQL.

**Comment by jdarwood007** [comment]
https://github.com/SimpleMachines/SMF2.1/commit/f02d980afcf5a2526a28bb3a9838c029ceb1b4fd#diff-ca7ff2a34b39d18aa0a1659d91368b9bR1827

Looks like Spuds.  Can we get this fixed up the proper way and use a join instead?

**Comment by Oldiesmann** [comment]
While we're on the subject of cleaning up queries, we should also clean up the conditional comments, since the only ones we use are for MySQL 4.x, which we no longer support (going by what we strip out of queries for PostgreSQL).

**Comment by tinoest** [comment]
@Oldiesmann did you want me to do that in this Pull Request also? As it's turned into a more a of a general fixes one.

File: Sources/ManageMaintenance.php [hunk 2]
```diff
@@ -1725,7 +1725,8 @@ function MaintainRecountPosts()
 	// Lets get a group of members and determine their post count (from the boards that have post count enabled of course).
 	$request = $smcFunc['db_query']('', '
 		SELECT /*!40001 SQL_NO_CACHE */ m.id_member, COUNT(m.id_member) AS posts
-		FROM ({db_prefix}messages AS m, {db_prefix}boards AS b)
+		FROM {db_prefix}messages AS m 
+		INNER JOIN {db_prefix}boards AS b ON m.id_board = b.id_board
 		WHERE m.id_member != {int:zero}
 			AND b.count_posts = {int:zero}
 			AND m.id_board = b.id_board ' . (!empty($modSettings['recycle_enable']) ? '
```
Comment (albertlast) [inline Sources/ManageMaintenance.php]: please remove here and in the other query the join statement  "AND m.id_board = b.id_board"
your already define the join statement in the ON part.

Comment (tinoest) [reply Sources/ManageMaintenance.php]: I left it in due to the !empty part, its already joined so I didn't see any performance impact.

File: Sources/ManageMaintenance.php [hunk 3]
```diff
@@ -1725,7 +1725,8 @@ function MaintainRecountPosts()
 	// Lets get a group of members and determine their post count (from the boards that have post count enabled of course).
 	$request = $smcFunc['db_query']('', '
 		SELECT /*!40001 SQL_NO_CACHE */ m.id_member, COUNT(m.id_member) AS posts
-		FROM ({db_prefix}messages AS m, {db_prefix}boards AS b)
+		FROM {db_prefix}messages AS m 
+		INNER JOIN {db_prefix}boards AS b ON m.id_board = b.id_board
```
Comment (Oldiesmann) [inline Sources/ManageMaintenance.php]: The INNER JOIN clause should be indented here

File: Sources/ManageMaintenance.php (hunk 2 above)
Comment (Oldiesmann) [reply Sources/ManageMaintenance.php]: I agree with @albertlast  here. No reason to leave this part in if we're doing an INNER JOIN.

**Pull request merged**